index/scorch/scorch.go

```diff
@@ -765,3 +787,53 @@ func parseToInteger(i interface{}) (int, error) {
 		return 0, fmt.Errorf("expects int or float64 value")
 	}
 }
+
+type fieldStats struct {
+	statMap map[string]map[string]uint64
+}
+
+func (fs *fieldStats) AddStat(statName, fieldName string, value uint64) {
+
+	if _, exists := fs.statMap[statName]; !exists {
+		fs.statMap[statName] = make(map[string]uint64)
+	}
+
+	fs.statMap[statName][fieldName] = value
+}
+
+func (fs *fieldStats) AggregateStats(stats segment.FieldStats) {
+
+	statMap := stats.GetStatsMap()
+	if statMap == nil {
+		return
+	}
+
+	for statName, statMap := range statMap {
+		for fieldName, val := range statMap {
+			fs.AppendStat(statName, fieldName, val)
+		}
+	}
+}
+
+func (fs *fieldStats) AppendStat(statName, fieldName string, value uint64) {
```

Comment:
I see no need for this exported API in the interface.

Reply:
Removed it completely